lib/blockchain/chaindb.js

```diff
@@ -1654,7 +1656,7 @@ class ChainDB {
       this.del(layout.p.encode(tip.hash));
       this.del(layout.h.encode(tip.hash));
       this.del(layout.e.encode(tip.hash));
-      this.del(layout.b.encode(tip.hash));
+      await this.blocks.prune(tip.hash);
```

Comment:
We lose atomicity here. We should consider the implications of a user hitting `^C`/`^\\` in the middle of this process. I wonder if it would be better to call `blocks.prune()` on everything only after the write batch has succeeded, but that would require a lot of logic rewrites.

Reply:
Hmm... `removeChains` is called exclusively by a chain `reset`, which from the documentation describes its use as replaying the blockchain download for SPV, and an SPV node doesn't doesn't store block as thus no need for a block store. Though `reset` can be called on any type of node.

The solution described works (and doesn't require too many changes) by queueing a set of block hashes in `_removeChain()` and returning the result back to `removeChains` to prune the blocks after a `commit()` per `tip`. It will prune blocks after alternative chain entries and records have been removed.

Reply:
Added